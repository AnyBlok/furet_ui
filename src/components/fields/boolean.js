// /**
// This file is a part of the FuretUI project
// 
//    Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>
// 
// This Source Code Form is subject to the terms of the Mozilla Public License,
// v. 2.0. If a copy of the MPL was not distributed with this file,You can
// obtain one at http://mozilla.org/MPL/2.0/.
// **/
// import _ from 'underscore';
import {defineComponent} from '../factory'
import {fields} from './fields';
import { listTemplate } from './common';


defineComponent('furet-ui-list-field-boolean', {
  template: `
    <div>
      <span v-if="isHidden" />
      <b-checkbox 
        v-else
        v-model="checked"
        disabled
      />
    </div>`,
  extend: ['furet-ui-list-field-common'],
  prototype: {
    computed: {
      checked () {
        return eval(this.value) ? true : false;
      },
    },
  },
})
fields.list.boolean = 'furet-ui-list-field-boolean'


defineComponent('furet-ui-list-field-yesno', {
  template: listTemplate,
  extend: ['furet-ui-list-field-common'],
  prototype: {
    computed: {
      value () {
        const base = 'components.fields.yesno'
        return this.$t(eval(this.data[this.config.name] || '') ? `${base}.yes` : `${base}.no`);
      },
    },
  },
})
fields.list.yesno = 'furet-ui-list-field-yesno'

//export const FieldThumbnailBoolean = Vue.component('furet-ui-thumbnail-field-boolean', {
//    mixins: [ThumbnailMixin],
//    template: `
//        <div v-if="this.isHidden" />
//        <b-tooltip 
//            v-bind:label="getTooltip" 
//            v-bind:position="tooltipPosition"
//            v-else
//        >
//            <b-checkbox 
//                v-model="value" 
//                disabled
//            >
//                {{this.label}}
//            </b-checkbox>
//        </b-tooltip>`,
//    computed: {
//        value () {
//            return eval(this.data[this.name]) ? true : false;
//        },
//    }
//})


defineComponent('furet-ui-form-field-boolean', {
  template: `
    <furet-ui-form-field-common-tooltip
      v-bind:data="data"
      v-bind:config="config"
    >
      <b-checkbox 
        v-model="checked" 
        v-bind:disabled="isReadonly"
      >
        {{ $t(config.label) }}
      </b-checkbox>
    </furet-ui-form-field-common-tooltip>`,
  extend: ['furet-ui-form-field-common'],
  prototype: {
    computed: {
      checked: {
        get () {
          return eval(this.value) ? true : false;
        },
        set (value) {
          this.updateValue (value);
        },
      },
    },
  },
})
fields.form.boolean = 'furet-ui-form-field-boolean'
