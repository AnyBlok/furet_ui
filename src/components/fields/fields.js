export const fields = {
  list: {},
  form: {},
  thumbnail: {},
  kanban: {},
};
export default {
    fields,
}
