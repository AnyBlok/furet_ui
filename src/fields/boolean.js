// /**
// This file is a part of the FuretUI project
// 
//    Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>
// 
// This Source Code Form is subject to the terms of the Mozilla Public License,
// v. 2.0. If a copy of the MPL was not distributed with this file,You can
// obtain one at http://mozilla.org/MPL/2.0/.
// **/
import Vue from 'vue';
import {FormMixin, ThumbnailMixin, ListMixin} from './common';
import _ from 'underscore';

export const FieldListBoolean = Vue.component('furet-ui-list-field-boolean', {
    mixins: [ListMixin],
    template: `
        <span v-if="isInvisible" />
        <b-checkbox 
            v-else
            v-bind:checked="checked" 
            disabled
        />`,
    computed: {
        checked () {
            return eval(this.value) ? true : false;
        },
    }
})

export const FieldThumbnailBoolean = Vue.component('furet-ui-thumbnail-field-boolean', {
    mixins: [ThumbnailMixin],
    template: `
        <div v-if="this.isInvisible" />
        <b-tooltip 
            v-bind:label="getTooltip" 
            v-bind:position="tooltipPosition"
            v-else
        >
            <b-checkbox 
                v-bind:checked="value" 
                disabled
            >
                {{this.label}}
            </b-checkbox>
        </b-tooltip>`,
    computed: {
        value () {
            return eval(this.data[this.name]) ? true : false;
        },
    }
})

export const FieldFormBoolean = Vue.component('furet-ui-form-field-boolean', {
    mixins: [FormMixin],
    template: `
        <div v-if="this.isInvisible" />
        <b-tooltip 
            v-bind:label="getTooltip" 
            v-bind:position="tooltipPosition"
            v-else
        >
            <b-checkbox 
                v-bind:checked="data" 
                v-bind:disabled="isReadonly"
            >
                {{this.label}}
            </b-checkbox>
        </b-tooltip>`,
    computed: {
        data () {
            const value = this.config && this.config.data && this.config.data[this.name] || '';
            return eval(value) ? true : false;
        },
    },
})
