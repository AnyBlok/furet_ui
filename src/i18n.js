/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import VueI18n from 'vue-i18n';
Vue.use(VueI18n);


export const i18nConf = {
    locale: 'en',
    fallbackLocale: 'en',
    messages: {
        en: {
            menus: {
                close: 'Close',
                search: 'Search ...',
            },
            views: {
                unknown: {
                    title: 'The wanted view "{name}" is unknown',
                    message: "Report this message to the administrator",
                },
                common: {
                    create: 'Create',
                    save: 'Save',
                    edit: 'Edit',
                    cancel: 'Cancel',
                    delete: 'Delete',
                    close: 'Close',
                    actions: 'Actions',
                    more: 'More',
                },
                clients: {
                    login: {
                        button: 'Log in',
                    },
                },
            },
            fields: {
                common: {
                    required: 'this field is required',
                },
                json: {
                    invalid: 'Invalid JSON format',
                },
                file: {
                    upload: 'Upload',
                    download: 'Download',
                    delete: 'Delete',
                },
            },
        },
    },
};
export const i18n = new VueI18n(i18nConf);
