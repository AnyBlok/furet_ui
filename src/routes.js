/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import VueRouter from 'vue-router';
Vue.use(VueRouter);
import {json_post_dispatch_all} from './server-call';


const CustomeView = {
    template: '<furet-ui-custom-view v-bind:viewName="viewName" />',
    props: ['viewName'],
    beforeRouteEnter: (to, from, next) => {
        json_post_dispatch_all('/custom/view/' + to.params.viewName, {});
        next();
    },
    beforeRouteUpdate: (to, from, next) => {
        if (to.params.viewName != from.params.viewName) {
            json_post_dispatch_all('/custom/view/' + to.params.viewName, {});
        }
        next();
    },
}

const Action = {
    template: '<router-view></router-view>',
    beforeRouteEnter: (to, from, next) => {
        json_post_dispatch_all('/action/' + to.params.actionId, to.params);
        next();
    },
    beforeRouteUpdate: (to, from, next) => {
        if (to.params.actionId != from.params.actionId)
            json_post_dispatch_all('/action/' + to.params.actionId, to.params);
        next();
    },
};

const View = {
    template: `<furet-ui-view
                    v-bind:spaceId="spaceId"
                    v-bind:menuId="menuId"
                    v-bind:actionId="actionId"
                    v-bind:viewId="viewId"
                    v-bind:dataId="dataId"
                    v-bind:mode="mode"
                />`,
    props: ['spaceId', 'menuId', 'actionId', 'viewId', 'dataId', 'mode'],
    beforeRouteEnter: (to, from, next) => {
        json_post_dispatch_all('/view/' + to.params.viewId, to.params);
        next();
    },
    beforeRouteUpdate: (to, from, next) => {
        if (to.params.viewId != from.params.viewId)
            json_post_dispatch_all('/view/' + to.params.viewId, to.params);
        next();
    },
};


export const router = new VueRouter({
    routes: [
        {
            path: '/custom/view/:viewName',
            name: 'custom_view',
            props: true,
            component: CustomeView,
        },
        {
            path: '/space/:spaceId',
            name: 'space',
            props: true,
            component: {
                template: `<furet-ui-space
                                v-bind:spaceId="spaceId"
                                v-bind:menuId="menuId"
                                v-bind:actionId="actionId"
                                v-bind:viewId="viewId"
                            />`,
                props: ['spaceId', 'menuId', 'actionId', 'viewId'],
                beforeRouteEnter: (to, from, next) => {
                    json_post_dispatch_all('/space/' + to.params.spaceId, to.params);
                    next();
                },
                beforeRouteUpdate: (to, from, next) => {
                    if (to.params.spaceId != from.params.spaceId) json_post_dispatch_all('/space/' + to.params.spaceId, to.params);
                    next();
                },
            },
            children: [
                {
                    path: 'menu/:menuId',
                    name: 'space_menu',
                    component: {
                        template: '<router-view></router-view>'
                    },
                    children: [
                        {
                            path: 'action/:actionId',
                            name: 'space_menu_action',
                            component: Action,
                            children: [
                                {
                                    path: 'view/:viewId',
                                    name: 'space_menu_action_view',
                                    props: true,
                                    component: View,
                                    children: [
                                        {
                                            path: 'data/:dataId/mode/:mode',
                                            name: 'space_menu_action_view_dataId',
                                            component: {template: '<div />'},  // never called
                                        },
                                    ],
                                },
                                {
                                    path: 'custom/view/:viewName',
                                    name: 'space_menu_action_custom_view',
                                    props: true,
                                    component: CustomeView,
                                },
                            ],
                        },
                    ],
                },
                {
                    path: 'action/:actionId',
                    name: 'space_action',
                    component: Action,
                    children: [
                        {
                            path: 'view/:viewId',
                            name: 'space_action_view',
                            props: true,
                            component: View,
                            children: [
                                {
                                    path: 'data/:dataId/mode/:mode',
                                    name: 'space_action_view_dataId',
                                    component: {template: '<div />'},  // never called
                                },
                            ],
                        },
                        {
                            path: 'custom/view/:viewName',
                            name: 'space_action_custom_view',
                            props: true,
                            component: CustomeView,
                        },
                    ],
                },
            ]
        },
    ],
});

export default router;
