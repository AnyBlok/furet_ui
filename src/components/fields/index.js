import { fields } from './fields';
import './common';
import './boolean';
import './datetime';
import './integer';
import './password';
import './timestamp';
import './selection';
import './string';

import './relationship';

import { defineComponent } from '../factory';

defineComponent('furet-ui-unknown-field', {
  template : `<div>Unknow Field <strong>{{config.type}}</strong></div>`,
  prototype: {
    props: ['config', 'data'],
  }
})

defineComponent('furet-ui-field', {
  template : `
    <component 
      v-bind:is="component" 
      v-bind:resource="resource" 
      v-bind:config="config" 
      v-bind:data="data"
    />
  `,
  prototype: {
    props: ['resource', 'config', 'data'],
    computed: {
      component () {
        return (fields[this.resource.type] || {})[this.config.type] || 'furet-ui-unknown-field'
      },
    },
  }
})
