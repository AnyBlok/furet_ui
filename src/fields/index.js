/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import './string';
import './selection';
import './datetime';
import './date';
import './time';
import './integer';
import './float';
import './url';
import './password';
import './color';
import './text';
import './boolean';
import './json';
import './relationship';
import './file';
