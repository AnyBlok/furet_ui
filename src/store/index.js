/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import _ from 'underscore';
import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from './actions';
import * as getters from './getters';
import modules from './modules';
import {i18n} from '../i18n';
import {router} from '../routes';
import moment from 'moment';

Vue.use(Vuex)
const debug = process.env.NODE_ENV !== 'production'

export const storeDef = {
    actions,
    getters,
    modules,
    strict: debug,
};

export const store = new Vuex.Store(storeDef);

export const dispatchAll = (datas) => {
    _.each(datas, data => {
        if (data.type) {
            switch (data.type) {
                case 'UPDATE_LOCALES':
                    _.each(data.locales, locale => {
                        i18n.setLocaleMessage(locale.locale, locale.messages)
                    });
                    break
                case 'SET_LOCALE':
                    i18n.locale = data.locale;
                    moment.locale(data.locale);
                    break
                case 'UPDATE_ROUTE':
                    if (data.name) router.push({name: data.name, params: data.params});
                    else if (data.path) router.push({path: data.path});
                    break;
                default:
                    const type = data.type;
                    delete data.type;
                    const process = data.process || 'commit';
                    store[process](type, data);
            }
        }
    });
}

export default store;
