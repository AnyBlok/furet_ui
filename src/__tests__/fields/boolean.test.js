/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import Vuex from 'vuex';
import Buefy from 'buefy';
Vue.use(Buefy, {defaultIconPack: 'fa',});
import {store} from '../../store';
import {router} from '../../routes';
import {i18n} from '../../i18n';
import {FieldListBoolean, FieldThumbnailBoolean, FieldFormBoolean} from '../../fields/boolean'

describe('Boolean list component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
    });
    it('Render is True', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListBoolean, {props: {
                row: {fieldname: true},
                header: {name: 'fieldname'},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render is False', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListBoolean, {props: {
                row: {fieldname: false},
                header: {name: 'fieldname'},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render without data', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListBoolean, {props: {
                row: {},
                header: {name: 'fieldname'},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is True', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListBoolean, {props: {
                row: {fieldname: false},
                header: {name: 'fieldname', invisible: true},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListBoolean, {props: {
                row: {fieldname: false},
                header: {name: 'fieldname', invisible: false},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is true', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListBoolean, {props: {
                row: {fieldname: false, invisible: true},
                header: {name: 'fieldname', invisible: 'fields.invisible'},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListBoolean, {props: {
                row: {fieldname: false, invisible: false},
                header: {name: 'fieldname', invisible: 'fields.invisible'},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});

describe('Boolean Thumbnail component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
    });
    it('Render is True', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailBoolean, {props: {
                data: {fieldname: true},
                name: 'fieldname',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render is False', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailBoolean, {props: {
                data: {fieldname: true},
                name: 'fieldname',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is True', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailBoolean, {props: {
                data: {fieldname: true},
                name: 'fieldname',
                invisible: true,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailBoolean, {props: {
                data: {fieldname: true},
                name: 'fieldname',
                invisible: false,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is true', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailBoolean, {props: {
                data: {fieldname: true, invisible: true},
                name: 'fieldname',
                invisible: 'fields.invisible',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailBoolean, {props: {
                data: {fieldname: true, invisible: false},
                name: 'fieldname',
                invisible: 'fields.invisible',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render label', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailBoolean, {props: {
                data: {fieldname: true},
                name: 'fieldname',
                label: 'The label',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render tooltip up', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailBoolean, {props: {
                data: {fieldname: true},
                name: 'fieldname',
                tooltip: 'The tooltip',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render tooltip left', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailBoolean, {props: {
                data: {fieldname: true},
                name: 'fieldname',
                tooltip: 'The left tooltip',
                tooltip_position: 'is-left',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});

describe('Boolean Form component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
    });
    it('Render is True', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormBoolean, {props: {
                config: {data: {fieldname: true}},
                name: 'fieldname',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render is False', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormBoolean, {props: {
                config: {data: {fieldname: true}},
                name: 'fieldname',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is True', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormBoolean, {props: {
                config: {data: {fieldname: true}},
                name: 'fieldname',
                invisible: true,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormBoolean, {props: {
                config: {data: {fieldname: true}},
                name: 'fieldname',
                invisible: false,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is true', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormBoolean, {props: {
                config: {data: {fieldname: true, invisible: true}},
                name: 'fieldname',
                invisible: 'fields.invisible',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormBoolean, {props: {
                config: {data: {fieldname: true, invisible: false}},
                name: 'fieldname',
                invisible: 'fields.invisible',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render label', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormBoolean, {props: {
                config: {data: {fieldname: true}},
                name: 'fieldname',
                label: 'The label',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render tooltip up', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormBoolean, {props: {
                config: {data: {fieldname: true}},
                name: 'fieldname',
                tooltip: 'The tooltip',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render tooltip left', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormBoolean, {props: {
                config: {data: {fieldname: true}},
                name: 'fieldname',
                tooltip: 'The left tooltip',
                tooltip_position: 'is-left',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render mode readonly by config', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormBoolean, {props: {
                config: {data: {fieldname: true}, mode: 'readonly'},
                name: 'fieldname',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render mode readonly param', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormBoolean, {props: {
                config: {data: {fieldname: true}},
                name: 'fieldname',
                readonly: true,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render mode readonly conditional param', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormBoolean, {props: {
                config: {data: {fieldname: true, readonly: 1}},
                name: 'fieldname',
                readonly: 'fields.readonly',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});
