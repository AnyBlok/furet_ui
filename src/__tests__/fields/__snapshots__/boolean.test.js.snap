// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boolean Form component Render invisible condition is false 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><label tabindex="0" class="checkbox"><input type="checkbox" true-value="true" checked="checked"> <span class="check"></span> <span class="control-label">
                
            </span></label>
</span>
`;

exports[`Boolean Form component Render invisible condition is true 1`] = `<div data-server-rendered="true"></div>`;

exports[`Boolean Form component Render invisible is True 1`] = `<div data-server-rendered="true"></div>`;

exports[`Boolean Form component Render invisible is false 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><label tabindex="0" class="checkbox"><input type="checkbox" true-value="true" checked="checked"> <span class="check"></span> <span class="control-label">
                
            </span></label>
</span>
`;

exports[`Boolean Form component Render is False 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><label tabindex="0" class="checkbox"><input type="checkbox" true-value="true" checked="checked"> <span class="check"></span> <span class="control-label">
                
            </span></label>
</span>
`;

exports[`Boolean Form component Render is True 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><label tabindex="0" class="checkbox"><input type="checkbox" true-value="true" checked="checked"> <span class="check"></span> <span class="control-label">
                
            </span></label>
</span>
`;

exports[`Boolean Form component Render label 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><label tabindex="0" class="checkbox"><input type="checkbox" true-value="true" checked="checked"> <span class="check"></span> <span class="control-label">
                The label
            </span></label>
</span>
`;

exports[`Boolean Form component Render mode readonly by config 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><label disabled="disabled" class="checkbox is-disabled"><input type="checkbox" disabled="disabled" true-value="true" checked="checked"> <span class="check"></span><span class="control-label">
                
            </span></label>
</span>
`;

exports[`Boolean Form component Render mode readonly conditional param 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><label disabled="disabled" class="checkbox is-disabled"><input type="checkbox" disabled="disabled" true-value="true" checked="checked"> <span class="check"></span><span class="control-label">
                
            </span></label>
</span>
`;

exports[`Boolean Form component Render mode readonly param 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><label disabled="disabled" class="checkbox is-disabled"><input type="checkbox" disabled="disabled" true-value="true" checked="checked"> <span class="check"></span><span class="control-label">
                
            </span></label>
</span>
`;

exports[`Boolean Form component Render tooltip left 1`] = `
<span data-label="The left tooltip" data-server-rendered="true" class="tooltip is-primary is-left is-medium"><label tabindex="0" class="checkbox"><input type="checkbox" true-value="true" checked="checked"> <span class="check"></span> <span class="control-label">
                
            </span></label>
</span>
`;

exports[`Boolean Form component Render tooltip up 1`] = `
<span data-label="The tooltip" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><label tabindex="0" class="checkbox"><input type="checkbox" true-value="true" checked="checked"> <span class="check"></span> <span class="control-label">
                
            </span></label>
</span>
`;

exports[`Boolean Thumbnail component Render invisible condition is false 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><label disabled="disabled" class="checkbox is-disabled"><input type="checkbox" disabled="disabled" true-value="true" checked="checked"> <span class="check"></span><span class="control-label">
                
            </span></label>
</span>
`;

exports[`Boolean Thumbnail component Render invisible condition is true 1`] = `<div data-server-rendered="true"></div>`;

exports[`Boolean Thumbnail component Render invisible is True 1`] = `<div data-server-rendered="true"></div>`;

exports[`Boolean Thumbnail component Render invisible is false 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><label disabled="disabled" class="checkbox is-disabled"><input type="checkbox" disabled="disabled" true-value="true" checked="checked"> <span class="check"></span><span class="control-label">
                
            </span></label>
</span>
`;

exports[`Boolean Thumbnail component Render is False 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><label disabled="disabled" class="checkbox is-disabled"><input type="checkbox" disabled="disabled" true-value="true" checked="checked"> <span class="check"></span><span class="control-label">
                
            </span></label>
</span>
`;

exports[`Boolean Thumbnail component Render is True 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><label disabled="disabled" class="checkbox is-disabled"><input type="checkbox" disabled="disabled" true-value="true" checked="checked"> <span class="check"></span><span class="control-label">
                
            </span></label>
</span>
`;

exports[`Boolean Thumbnail component Render label 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><label disabled="disabled" class="checkbox is-disabled"><input type="checkbox" disabled="disabled" true-value="true" checked="checked"> <span class="check"></span><span class="control-label">
                The label
            </span></label>
</span>
`;

exports[`Boolean Thumbnail component Render tooltip left 1`] = `
<span data-label="The left tooltip" data-server-rendered="true" class="tooltip is-primary is-left is-medium"><label disabled="disabled" class="checkbox is-disabled"><input type="checkbox" disabled="disabled" true-value="true" checked="checked"> <span class="check"></span><span class="control-label">
                
            </span></label>
</span>
`;

exports[`Boolean Thumbnail component Render tooltip up 1`] = `
<span data-label="The tooltip" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><label disabled="disabled" class="checkbox is-disabled"><input type="checkbox" disabled="disabled" true-value="true" checked="checked"> <span class="check"></span><span class="control-label">
                
            </span></label>
</span>
`;

exports[`Boolean list component Render invisible condition is false 1`] = `<label disabled="disabled" data-server-rendered="true" class="checkbox is-disabled"><input type="checkbox" disabled="disabled" true-value="true"> <span class="check"></span> <span class="control-label"></span></label>`;

exports[`Boolean list component Render invisible condition is true 1`] = `<span data-server-rendered="true"></span>`;

exports[`Boolean list component Render invisible is True 1`] = `<span data-server-rendered="true"></span>`;

exports[`Boolean list component Render invisible is false 1`] = `<label disabled="disabled" data-server-rendered="true" class="checkbox is-disabled"><input type="checkbox" disabled="disabled" true-value="true"> <span class="check"></span> <span class="control-label"></span></label>`;

exports[`Boolean list component Render is False 1`] = `<label disabled="disabled" data-server-rendered="true" class="checkbox is-disabled"><input type="checkbox" disabled="disabled" true-value="true"> <span class="check"></span> <span class="control-label"></span></label>`;

exports[`Boolean list component Render is True 1`] = `<label disabled="disabled" data-server-rendered="true" class="checkbox is-disabled"><input type="checkbox" disabled="disabled" true-value="true" checked="checked"> <span class="check"></span> <span class="control-label"></span></label>`;

exports[`Boolean list component Render without data 1`] = `<label disabled="disabled" data-server-rendered="true" class="checkbox is-disabled"><input type="checkbox" disabled="disabled" true-value="true"> <span class="check"></span> <span class="control-label"></span></label>`;
